src/gazebo_mavlink_interface.cpp

```diff
@@ -850,72 +854,77 @@ void GazeboMavlinkInterface::SendSensorMessages()
     last_imu_message_.orientation().y(),
     last_imu_message_.orientation().z());
 
-  ignition::math::Quaterniond q_gb = q_gr*q_br.Inverse();
-  ignition::math::Quaterniond q_nb = q_ng*q_gb;
-
+  bool should_send_imu = false;
+  if (!enable_lockstep_) {
 #if GAZEBO_MAJOR_VERSION >= 9
-  ignition::math::Vector3d pos_g = model_->WorldPose().Pos();
+    common::Time current_time = world_->SimTime();
 #else
-  ignition::math::Vector3d pos_g = ignitionFromGazeboMath(model_->GetWorldPose().pos);
+    common::Time current_time = world_->GetSimTime();
 #endif
-  ignition::math::Vector3d pos_n = q_ng.RotateVector(pos_g);
+    double dt = (current_time - last_imu_time_).Double();
+
+    if (imu_update_interval_!=0 && dt >= imu_update_interval_) {
+      should_send_imu = true;
+      last_imu_time_ = current_time;
+    }
+  }
 
+  mavlink_hil_sensor_t sensor_msg;
 #if GAZEBO_MAJOR_VERSION >= 9
-  ignition::math::Vector3d vel_b = q_br.RotateVector(model_->RelativeLinearVel());
-  ignition::math::Vector3d vel_n = q_ng.RotateVector(model_->WorldLinearVel());
-  ignition::math::Vector3d omega_nb_b = q_br.RotateVector(model_->RelativeAngularVel());
+  sensor_msg.time_usec = world_->SimTime().Double() * 1e6;
 #else
-  ignition::math::Vector3d vel_b = q_br.RotateVector(ignitionFromGazeboMath(model_->GetRelativeLinearVel()));
-  ignition::math::Vector3d vel_n = q_ng.RotateVector(ignitionFromGazeboMath(model_->GetWorldLinearVel()));
-  ignition::math::Vector3d omega_nb_b = q_br.RotateVector(ignitionFromGazeboMath(model_->GetRelativeAngularVel()));
+  sensor_msg.time_usec = world_->GetSimTime().Double() * 1e6;
 #endif
 
+  // send always accel and gyro data (not dependent of the bitmask)
+  // required so to keep the timestamps on sync and the lockstep can
+  // work properly
```

Comment:
I'm just irritated because there is an entry for gyro in the bitmask but none for accel.

Reply:
https://mavlink.io/en/messages/common.html#HIL_SENSOR.